libnmstate/nm/connection.py

```diff
@@ -249,34 +258,77 @@ def _get_activation_metadata(self):
 
         return activation_type, activation_object
 
-    def waitfor_active_connection_async(self, ac):
-        ac.handlers.add(
-            ac.nm_active_connection.connect(
-                'state-changed', self._waitfor_active_connection_callback, ac
+    def wait_active_async(self):
```

Comment:
I think it will be more useful (and a bit more readable for me at least), to name and place it in the mainloop:
```
self._mainloop.async_sleep(timeout=ACTIVATE_CHECK_INTERNAL, timeout_func=self._waitfor_active_connection_callback)
```

Reply:
I intend to keep the function close to its caller. We can move to mainloop if other file/module need this kind of function.